eclass/cmake.eclass

```diff
@@ -363,13 +363,6 @@ cmake_src_prepare() {
 		die "FATAL: Unable to find CMakeLists.txt"
 	fi
 
-	# if ninja is enabled but not installed, the build could fail
-	# this could happen if ninja is manually enabled (eg. make.conf) but not installed
-	if [[ ${CMAKE_MAKEFILE_GENERATOR} == ninja ]] && ! has_version -b dev-util/ninja; then
```

Comment:
Why are we dropping this again?

Reply:
I received this reply on the ML from Mike Gilbert.
```
>  	# this could happen if ninja is manually enabled (eg. make.conf) but not installed  
-	if [[ ${CMAKE_MAKEFILE_GENERATOR} == ninja ]] && ! has_version -b dev-util/ninja; then
+	if [[ ${CMAKE_MAKEFILE_GENERATOR} == ninja ]] && [[ -n "${NINJA_DEPEND}" ]] && ! has_version -b "${NINJA_DEPEND}"; then

I would just drop this check. BDEPEND should suffice to ensure that ninja or samurai is installed.
```
I can add it back if desired, it doesn't really matter to me.

Reply:
Oh, sorry, I missed that. No, go wild.

Reply:
That snippet was introduced to cmake-utils.eclass by commit e9cd2c48c36ac18a3b50ab09e1ced545260d4b30.

It helps covering for Portage failing to build things in the right order: https://bugs.gentoo.org/561570

Reply:
Bug [561570](https://bugs.gentoo.org/561570) was caused by toralf setting CMAKE_MAKEFILE_GENERATOR in make.conf, not by some bug in Portage.

Setting CMAKE_MAKEFILE_GENERATOR in make.conf violates the metadata invariance rule, and users must ensure they install the necessary packages manually in that case.

I don't think this has_version check in the eclass is wanted or necessary.